zipline/errors.py

```diff
@@ -433,10 +433,21 @@ class TermInputsNotSpecified(ZiplineError):
     msg = "{termname} requires inputs, but no inputs list was passed."
 
 
+class TermOutputsNotSpecified(ZiplineError):
+    """
+    Raised if a user attempts to construct a CustomFactor without specifying
+    enough outputs and that factor does not have class-level default outputs.
+    """
+    msg = (
+        "{termname} requires at least two outputs, but was given "
```

Comment:
Ah ok. So for the purpose of RecarrayFactors as we have them it makes sense, but wouldn't hurt to be more generalizable in case we come across such a case.